pkg/controllers/status/cluster_status_controller.go

```diff
@@ -229,13 +233,22 @@ func (c *ClusterStatusController) setCurrentClusterStatus(clusterClient *util.Cl
 	}
 	currentClusterStatus.KubernetesVersion = clusterVersion
 
+	var apiEnablementCondition metav1.Condition
 	// get the list of APIs installed in the member cluster
 	apiEnables, err := getAPIEnablements(clusterClient)
 	if len(apiEnables) == 0 {
+		apiEnablementCondition = util.NewCondition(clusterv1alpha1.ClusterConditionInvalidAPIEnablements,
+			apiEnablementEmptyAPIEnablements, "collected empty APIEnablements from the cluster", metav1.ConditionTrue)
 		klog.Errorf("Failed to get any APIs installed in Cluster %s. Error: %v.", cluster.GetName(), err)
 	} else if err != nil {
+		apiEnablementCondition = util.NewCondition(clusterv1alpha1.ClusterConditionInvalidAPIEnablements,
+			apiEnablementPartialAPIEnablements, fmt.Sprintf("might collect partial APIEnablements(%d) from the cluster", len(apiEnables)), metav1.ConditionTrue)
 		klog.Warningf("Maybe get partial(%d) APIs installed in Cluster %s. Error: %v.", len(apiEnables), cluster.GetName(), err)
+	} else {
+		apiEnablementCondition = util.NewCondition(clusterv1alpha1.ClusterConditionInvalidAPIEnablements,
+			apiEnablementSyncSucceed, "collected full APIEnablements from the cluster", metav1.ConditionFalse)
 	}
+	meta.SetStatusCondition(&currentClusterStatus.Conditions, apiEnablementCondition)
```

Comment:
It may not be possible to set this directly. I just recently fixed a bug related to this, ref #5227